python/lsst/ip/isr/photodiode.py

```diff
@@ -0,0 +1,260 @@
+                               f"found {dictionary['metadata']['OBSTYPE']}")
+
+        calib.setMetadata(dictionary['metadata'])
+
+        calib.timeSamples = np.array(dictionary.get('timeSamples'))
+        calib.currentSamples = np.array(dictionary.get('currentSamples'))
+        calib.integrationMethod = dictionary.get('integrationMethod')
+
+        calib.updateMetadata()
+        return calib
+
+    def toDict(self):
+        """Return a dictionary containing the photodiode properties.
+
+        The dictionary should be able to be round-tripped through.
+        `fromDict`.
+
+        Returns
+        -------
+        dictionary : `dict`
+            Dictionary of properties.
+        """
+        self.updateMetadata()
+
+        outDict = {}
+        metadata = self.getMetadata()
+        outDict['metadata'] = metadata
+
+        outDict['timeSamples'] = self.timeSamples.tolist()
+        outDict['currentSamples'] = self.currentSamples.tolist()
+
+        outDict['integrationMethod'] = self.integrationMethod
+
+        return outDict
+
+    @classmethod
+    def fromTable(cls, tableList):
+        """Construct calibration from a list of tables.
+
+        This method uses the `fromDict` method to create the
+        calibration after constructing an appropriate dictionary from
+        the input tables.
+
+        Parameters
+        ----------
+        tableList : `list` [`astropy.table.Table`]
+            List of tables to use to construct the crosstalk
+            calibration.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.PhotodiodeCalib`
+            The calibration defined in the tables.
+        """
+        dataTable = tableList[0]
+
+        metadata = dataTable.meta
+        inDict = {}
+        inDict['metadata'] = metadata
+        inDict['integrationMethod'] = metadata.pop('INTEGRATION_METHOD', None)
+
+        inDict['timeSamples'] = dataTable['TIME']
+        inDict['currentSamples'] = dataTable['CURRENT']
+
+        return cls().fromDict(inDict)
+
+    def toTable(self):
+        """Construct a list of tables containing the information in this
+        calibration.
+
+        The list of tables should create an identical calibration
+        after being passed to this class's fromTable method.
+
+        Returns
+        -------
+        tableList : `list` [`astropy.table.Table`]
+            List of tables containing the photodiode calibration
+            information.
+        """
+        self.updateMetadata()
+        catalog = Table([{'TIME': self.timeSamples,
+                          'CURRENT': self.currentSamples}])
+        inMeta = self.getMetadata().toDict()
+        outMeta = {k: v for k, v in inMeta.items() if v is not None}
+        outMeta.update({k: "" for k, v in inMeta.items() if v is None})
+        outMeta['INTEGRATION_METHOD'] = self.integrationMethod
+        catalog.meta = outMeta
+
+        return([catalog])
```

Comment:
I imagine there's a much bigger picture reason why this returns something of this shape... Is this so that it's in keeping with other `calib.toTable()` methods?